### PR TITLE
Add Browser support for using Cookies

### PR DESCRIPTION
**Pull request by aroberson**
Added a property for browser to set the withCredentials flag on the XMLHttpRequest for upload and download

### CODE REVIEW

File: www/browser/FileTransfer.js [hunk 1]
```diff
@@ -121,6 +122,9 @@ FileTransfer.prototype.upload = function(filePath, server, successCallback, erro
 
     var that = this;
     var xhr = transfers[this._id] = new XMLHttpRequest();
+    if (withCredentials) {
+        xhr.withCredentials = true;
```
Comment (dukex) [inline www/browser/FileTransfer.js]: You can remove the `if`statement and write only:
``` js 
 xhr.withCredentials = withCredentials
```

File: www/browser/FileTransfer.js [hunk 2]
```diff
@@ -231,7 +236,9 @@ FileTransfer.prototype.download = function(source, target, successCallback, erro
 
     var that = this;
     var xhr = transfers[this._id] = new XMLHttpRequest();
-    
+    if (withCredentials) {
+        xhr.withCredentials = true;
```
Comment (dukex) [inline www/browser/FileTransfer.js]: Same here

**Comment by purplecabbage** [comment]
Is 'withCredentials' available everywhere? I think this has potentially wider impact than just the cordova-browser platform.
Shouldn't we check if xhr.withCredentials exists?
Then this could become : 
``` 
if('withCredentials' in xhr) {
  xhr.withCredentials = options.withCredentials || false;
}
```

**Comment by aroberson** [comment]
Duke,
I did not do that on purpose as I wanted to keep it as close to possible to the original functionality.  I can make the change.

I would have to check the other platforms.  Most of them already handle the use of cookies properly which is why I did not apply the change elsewhere.

**Pull request closed without merging**